pkg/volume/util/operationexecutor/operation_generator.go

```diff
@@ -820,6 +823,54 @@ func (og *operationGenerator) GenerateVerifyControllerAttachedVolumeFunc(
 	}, nil
 }
 
+func (og *operationGenerator) GenerateVerifyControllerDetachedVolumeFunc(
+	volumeToDetach AttachedVolume,
+	actualStateOfWorld ActualStateOfWorldAttacherUpdater) (func() error, error) {
+	return func() error {
+		if !volumeToDetach.PluginIsAttachable {
+			// If the volume does not implement the attacher interface, it is
+			// assumed to be detached and the actual state of the world is
+			// updated accordingly.
+			actualStateOfWorld.MarkVolumeAsDetached(volumeToDetach.VolumeName, volumeToDetach.NodeName)
+			return nil
+		}
+
+		// Fetch current node object
+		node, fetchErr := og.kubeClient.Core().Nodes().Get(string(volumeToDetach.NodeName), metav1.GetOptions{})
```

Comment:
I think this is a good solution by putting this function in to operation_executor. But I feel this Fetch current node object might cause unnecessary overhead by querying API server frequently. It is ok to remove this fetch node object part and just call MarkVolumeAsDetached() function. The reason is that before calling this  VerifyControllerDetachedVolume(), reconcilder already checked that attachedVolume.GloballyMounted is false. This means that there is no global mount (mount a device path to a global path) exists or there is a pending operation for this volume being mounted. However, if there is an pending operation for this volume, operation_executor will not start this   VerifyControllerDetachedVolume() at all. It must wait until the pending operation for this volume finishes, it can start this verify detached volume operation.